ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/Node.hs

```diff
@@ -107,26 +108,32 @@ instance TPraosCrypto era => SerialiseConstraintsHFC (ShelleyBlock era)
 -- 0. Byron EBB
 -- 1. Byron regular block
 -- 2. Shelley block
--- 3. Goguen block
--- 4. etc.
+-- 3. Allegra block
+-- 4. Mary block
+-- 5. Goguen block
+-- 6. etc.
 --
 -- For more details, see:
 -- <https://github.com/input-output-hk/ouroboros-network/pull/1175#issuecomment-558147194>
 instance CardanoHardForkConstraints c => SerialiseHFC (CardanoEras c) where
-  encodeDiskHfcBlock (CardanoCodecConfig ccfgByron ccfgShelley) = \case
+  encodeDiskHfcBlock (CardanoCodecConfig ccfgByron ccfgShelley ccfgAllegra ccfgMary) = \case
       -- We are backwards compatible with Byron and thus use the exact same
       -- encoding.
       BlockByron   blockByron   ->                encodeDisk ccfgByron blockByron
-      -- For Shelley, we need to prepend the hard fork envelope.
+      -- For Shelley and later eras, we need to prepend the hard fork envelope.
       BlockShelley blockShelley -> prependTag 2 $ encodeDisk ccfgShelley blockShelley
-  decodeDiskHfcBlock (CardanoCodecConfig ccfgByron ccfgShelley) = do
+      BlockAllegra blockAllegra -> prependTag 3 $ encodeDisk ccfgAllegra blockAllegra
+      BlockMary    blockMary    -> prependTag 4 $ encodeDisk ccfgMary    blockMary
```

Comment:
This is slightly different from the other encoders, which throw an exception for these cases when the version doesn't yet enable these eras.